packages/dev/gui/src/2D/controls/container.ts

```diff
@@ -619,6 +621,42 @@ export class Container extends Control {
         this._measureForChildren.copyFrom(this._currentMeasure);
     }
 
+    public isWidthFullyDefined(): boolean {
+        if (!this._isDirty && this._cachedIsWidthFullyDefined !== undefined) {
+            return this._cachedIsWidthFullyDefined;
+        }
+        if (this.adaptWidthToChildren) {
+            for (const child of this.children) {
+                if (!child.isWidthFullyDefined()) {
+                    this._cachedIsWidthFullyDefined = false;
+                    return false;
+                }
+            }
+            this._cachedIsWidthFullyDefined = true;
+            return true;
+        }
+        this._cachedIsWidthFullyDefined = super.isWidthFullyDefined();
+        return this._cachedIsWidthFullyDefined;
+    }
+
+    public isHeightFullyDefined(): boolean {
+        if (!this._isDirty && this._cachedIsHeightFullyDefined !== undefined) {
+            return this._cachedIsHeightFullyDefined;
+        }
+        if (this.adaptHeightToChildren) {
+            for (const child of this.children) {
+                if (!child.isHeightFullyDefined()) {
+                    this._cachedIsHeightFullyDefined = false;
+                    return false;
+                }
+            }
+            this._cachedIsHeightFullyDefined = true;
+            return true;
+        }
+        this._cachedIsHeightFullyDefined = super.isHeightFullyDefined();
+        return this._cachedIsHeightFullyDefined;
+    }
+

```

Comment:
those two functions (later as well) are identical, with the only difference is height vs. width. Is there a way to reuse the code with a different parameter set?

Reply:
@RaananW great observation, I adapted the function so that I can pass the dimension to check and it does the process for the selected dimension.